tools/pythonpkg/src/include/duckdb_python/pyconnection.hpp

```diff
@@ -24,6 +24,7 @@ struct DuckDBPyConnection {
 	shared_ptr<DuckDB> database;
 	unique_ptr<Connection> connection;
 	unordered_map<string, py::object> registered_dfs;
+	unordered_map<string, py::object> registered_arrow;
```

Comment:
Perhaps we can handle this in a nicer way, e.g. have an abstract class `RegisteredObject` here, that can be overloaded for the DataFrame and the Arrow object separately:

```cpp
class RegisteredObject {
public:
   RegisteredObject(py::obj obj_p) : obj(move(obj_p)) {}
   virtual ~RegisteredObject(){}

   py::obj obj;

public:
   virtual void Unregister() = 0;
};
```

Then we can use a single unordered map here (and potentially unregistering could be done automatically on destruction?).